### PR TITLE
Fix Helios auto offload decode

### PR DESCRIPTION
**Pull request by pzarzycki**
## Summary

Fixes `HeliosDecodeStep` when modular Helios pipelines run with component auto CPU offload enabled.

The decode step was normalizing chunk latents using `latents_mean` / `latents_std` tensors created on the VAE device, while the chunk latents themselves could remain on a different device under offload. That caused device mismatches during decode.

This change aligns Helios with the working `Wan` decoder pattern by normalizing each chunk on the chunk latent tensor's own device and dtype before converting to the VAE decode dtype.

## Repro

```bash
python -m pytest tests/modular_pipelines/helios/test_modular_pipeline_helios.py::TestHeliosModularPipelineFast::test_components_auto_cpu_offload_inference_consistent -q
```

Before this change, the test failed in `src/diffusers/modular_pipelines/helios/decoders.py` with a device mismatch during decode under auto CPU offload.

## Implementation

- updated `HeliosDecodeStep` in `src/diffusers/modular_pipelines/helios/decoders.py`
- compute `latents_mean` / `latents_std` on each chunk latent tensor's device and dtype
- normalize chunk latents there first, then cast to the VAE decode dtype
- keep the rest of the decode flow unchanged

## Tests

Ran on accelerator-backed setup:

```bash
python -m pytest tests/modular_pipelines/helios/test_modular_pipeline_helios.py::TestHeliosModularPipelineFast::test_components_auto_cpu_offload_inference_consistent -q
python -m pytest tests/modular_pipelines/helios/test_modular_pipeline_helios.py -q
```

Results:

```text
1 passed
24 passed, 4 skipped
```

Also ran CPU-mode validation:

```bash
DIFFUSERS_TEST_DEVICE=cpu python -m pytest tests/modular_pipelines/helios/test_modular_pipeline_helios.py -q
```

Result:

```text
17 passed, 11 skipped
```

### CODE REVIEW

File: src/diffusers/modular_pipelines/helios/decoders.py [hunk 1]
```diff
@@ -77,16 +77,18 @@ def __call__(self, components, state: PipelineState) -> PipelineState:
...
-        )
-
         history_video = None
         for chunk_latents in block_state.latent_chunks:
-            current_latents = chunk_latents.to(vae.dtype) / latents_std + latents_mean
+            latents_mean = (
+                torch.tensor(vae.config.latents_mean).view(1, vae.config.z_dim, 1, 1, 1).to(
+                    chunk_latents.device, chunk_latents.dtype
+                )
+            )
```
[the remark this answers is not in the document]
Comment (yiyixuxu) [reply src/diffusers/modular_pipelines/helios/decoders.py]: Can we just use this? 

```py
device = components._exeution_device
```

[the remark this answers is not in the document]
Comment (pzarzycki) [reply src/diffusers/modular_pipelines/helios/decoders.py]: I updated this to use `components._execution_device` in `HeliosDecodeStep` and moved the `latents_mean` / `latents_std` setup out of the chunk loop.

Using `chunk_latents.device` fixes the mismatch, but it also rebuilds those tensors for every chunk. Since the modular pipeline already uses `components._execution_device` as the execution source of truth in nearby Helios steps, that seemed like the cleaner fix here.

I also added a regression test for the offload case where `vae.device` is CPU but decode runs on the execution device.

I reran the full Helios modular test file after that:

- `DIFFUSERS_TEST_DEVICE=cpu .../test_modular_pipeline_helios.py` -> 17 passed, 12 skipped
- `DIFFUSERS_TEST_DEVICE=cuda .../test_modular_pipeline_helios.py` -> 25 passed, 4 skipped

File: tests/modular_pipelines/helios/test_modular_pipeline_helios.py [hunk 2]
```diff
@@ -91,6 +97,51 @@ def test_num_images_per_prompt(self):
         pass
 
 
+class DummyHeliosVAE:
```
Comment (yiyixuxu) [inline tests/modular_pipelines/helios/test_modular_pipeline_helios.py]: can you remove the test? we already have offloading test https://github.com/huggingface/diffusers/blob/main/tests/modular_pipelines/test_modular_pipelines_common.py#L350

**Comment by pzarzycki** [comment]
Removed the Helios-specific offload regression test as suggested.

I kept the decoder fix and dropped the redundant test/helper block in `tests/modular_pipelines/helios/test_modular_pipeline_helios.py`, since the shared modular offloading coverage in `tests/modular_pipelines/test_modular_pipelines_common.py` already covers that behavior.

I also fixed the pending formatter issue on `src/diffusers/modular_pipelines/helios/decoders.py` so the branch passes the quality check.

Local validation:
- `pytest tests/modular_pipelines/helios/test_modular_pipeline_helios.py -q` -> `24 passed, 4 skipped`
- `ruff check examples scripts src tests utils benchmarks setup.py`
- `ruff format --check examples scripts src tests utils benchmarks setup.py`

**Review by yiyixuxu — APPROVED** [review]
thanks
